### DISCUSSION TITLE
Message: Uncaught TypeError: Cannot read property 'PrdCodPlanprod' of undefined

### DISCUSSION THREAD

**Discussion by patriciomh1965** [serenity-is/Serenity#5668] [Help]
Hello community

I have the following problem that I have fixed. I have searched the issues: I present the sentences

namespace PLAINT.ServiciosProducto {

    @Serenity.Decorators.registerClass()
    @Serenity.Decorators.panel()
    export class TaCncPlanesDialog extends Serenity.EntityDialog<TaCncPlanesRow, any> {
        protected getFormKey() { return TaCncPlanesForm.formKey; }
        protected getIdProperty() { return TaCncPlanesRow.idProperty; }
        protected getLocalTextPrefix() { return TaCncPlanesRow.localTextPrefix; }
        protected getNameProperty() { return TaCncPlanesRow.nameProperty; }
        protected getService() { return TaCncPlanesService.baseUrl; }
        protected getDeletePermission() { return TaCncPlanesRow.deletePermission; }
        protected getInsertPermission() { return TaCncPlanesRow.insertPermission; }
        protected getUpdatePermission() { return TaCncPlanesRow.updatePermission; }

        protected form = new TaCncPlanesForm(this.idPrefix);

        //  SACAR BOTONES
        protected getToolbarButtons(): Serenity.ToolButton[] {
            let buttons = super.getToolbarButtons();
            buttons.splice(Q.indexOf(buttons, x => x.cssClass == "apply-changes-button"), 1);
            return buttons;
        }

        protected updateTitle(): void {
            super.updateTitle();
            this.dialogTitle = 'Planes';
        }

         protected afterLoadEntity() {
            super.afterLoadEntity();

             Serenity.EditorUtils.setReadOnly(this.form.PlnPlataforma, true);
             Serenity.EditorUtils.setReadOnly(this.form.PlnEquiNumero, true);
             Serenity.EditorUtils.setReadOnly(this.form.PlnCodPlanProd, true);                          
        }

        constructor()
        {
            super();

            this.form = new TaCncPlanesForm(this.idPrefix);
            Serenity.SubDialogHelper.triggerDataChange(this);

            this.form.PlnPlanProducto.change(() =>
            {
                var planproductoID = Q.toId(this.form.PlnPlanProducto.value);               
                if (planproductoID != 0)
                {                    
                  //  Q.notifyInfo('planproductoID ' + planproductoID);
                    var codplanID = TaCncParamPlanProductoDatosRow.getLookup().itemById[this.form.PlnPlanProducto.value].PrdCodPlanprod;
                    Q.notifyInfo('PrdCodPlanprod ' + codplanID);
                }
            });

        }
    }
}

![error](https://user-images.githubusercontent.com/42218266/116358524-c0379580-a7cb-11eb-9be0-e317426c53d3.png)

It does not recognize the PrdCodPlanprod field, however it exists in the class:

namespace PLAINT.ServiciosProducto.Entities
{
    using Serenity;
    using Serenity.ComponentModel;
    using Serenity.Data;
    using Serenity.Data.Mapping;
    using System;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.IO;

    [ConnectionKey("ServiciosProducto"), Module("ServiciosProducto"), TableName("[dbo].[ta_cnc_param_plan_producto]")]
    [DisplayName("Plan Producto"), InstanceName("Plan Producto")]
    [ReadPermission("Administration:General")]
    [ModifyPermission("Administration:General")]

    [LookupScript(Permission = "*")] // VINCULACION
    public sealed class TaCncParamPlanProductoDatosRow : Row, IIdRow
    {
        [DisplayName("Numero"), Column("prd_numero"), Identity]
        public Int32? PrdNumero
        {
            get { return Fields.PrdNumero[this]; }
            set { Fields.PrdNumero[this] = value; }
        }
        [DisplayName("Cod Planprod"), Column("prd_cod_planprod"), LookupInclude]
        public Int32? PrdCodPlanprod
        {
            get { return Fields.PrdCodPlanprod[this]; }
            set { Fields.PrdCodPlanprod[this] = value; }
        }

        IIdField IIdRow.IdField
        {
            get { return Fields.PrdNumero; }
        }

     

        public static readonly RowFields Fields = new RowFields().Init();

        public TaCncParamPlanProductoDatosRow() 
            : base(Fields)
        {
        }

        public class RowFields : RowFieldsBase
        {
            public Int32Field PrdNumero;
            public Int32Field PrdCodPlanprod;

        }
    }
}

Someone who can tell me where my mistake is, I would appreciate it.

Atte.

**Comment by minhhungit** [comment]
You need to check the field on your Form.cs, not row.cs, and make sure you tranformmed t4 template

**Comment by patriciomh1965** [comment]
Hi jin The PrdCodPlanprod property exists in the form, reading some cases
of the error some people in the community indicate that it may be a problem
with the installation of the NuGet package. The truth is that I have not
been able to locate the problem. I would like an orientation since you are
a more suitable professional. Apologize for so many annoyances. Atte.
Patrick

**Reply by minhhungit** [reply]
I did not check it carefully, this line cause your error:
```javascritp
var codplanID = TaCncParamPlanProductoDatosRow.getLookup().itemById[this.form.PlnPlanProducto.value].PrdCodPlanprod;
Q.notifyInfo('PrdCodPlanprod ' + codplanID);
```

It is not problem about nuget package

Problem is this code return null value:

```javascript
 TaCncParamPlanProductoDatosRow.getLookup().itemById[this.form.PlnPlanProducto.value]
```


you should make sure that the lookup contains value `this.form.PlnPlanProducto.value`

Normally the code should be re-write like this:

```javascript
var existedItem = TaCncParamPlanProductoDatosRow.getLookup().itemById[this.form.PlnPlanProducto.value];
if (existedItem != null && typeof(existedItem) != 'undefined')
{
    Q.notifyInfo('PrdCodPlanprod ' + existedItem.PrdCodPlanprod);
}
```

**Comment by patriciomh1965** [comment]
Jin

Thank you very much, your help was great my friend

Atte.




namespace PLAINT.ServiciosProducto {

    @Serenity.Decorators.registerClass()
    @Serenity.Decorators.panel()
    export class TaCncPlanesDialog extends
Serenity.EntityDialog<TaCncPlanesRow, any> {
        protected getFormKey() { return TaCncPlanesForm.formKey; }
        protected getIdProperty() { return TaCncPlanesRow.idProperty; }
        protected getLocalTextPrefix() { return
TaCncPlanesRow.localTextPrefix; }
       // protected getNameProperty() { return TaCncPlanesRow.nameProperty;
}
        protected getService() { return TaCncPlanesService.baseUrl; }
     //   protected getDeletePermission() { return
TaCncPlanesRow.deletePermission; }
     //   protected getInsertPermission() { return
TaCncPlanesRow.insertPermission; }
     //   protected getUpdatePermission() { return
TaCncPlanesRow.updatePermission; }

        protected form = new TaCncPlanesForm(this.idPrefix);

        //  SACAR BOTONES
        protected getToolbarButtons(): Serenity.ToolButton[] {
            let buttons = super.getToolbarButtons();
            buttons.splice(Q.indexOf(buttons, x => x.cssClass ==
"apply-changes-button"), 1);
            return buttons;
        }


        protected updateTitle(): void {
            super.updateTitle();
            this.dialogTitle = 'Planes';
        }

         protected afterLoadEntity() {
            super.afterLoadEntity();

             Serenity.EditorUtils.setReadOnly(this.form.PlnPlataforma,
true);
             Serenity.EditorUtils.setReadOnly(this.form.PlnCodPlanProd,
true);
        }

        constructor()
        {
            super();

            this.form = new TaCncPlanesForm(this.idPrefix);
            Serenity.SubDialogHelper.triggerDataChange(this);

            this.form.PlnPlanProducto.change(() =>
            {
                var codplanprodID = 0;
                var planproductoID =
Q.toId(this.form.PlnPlanProducto.value);
                if (planproductoID != 0)
                {





* var existedItem =
TaCncParamPlanProductoDatosRow.getLookup().itemById[this.form.PlnPlanProducto.value];
                  if (existedItem != null && typeof (existedItem) !=
'undefined') {                        codplanprodID =
existedItem.PrdCodPlanProd;                     //
Q.notifyInfo('PrdCodPlanprod ' + existedItem.PrdCodPlanProd);
      }                    this.form.PlnCodPlanProd.value = codplanprodID;*
                }
            });

        }
    }
}
